perf: Double speed of interpreter serialization

NeilFraser opened the pull request:
Bumps our copies of JS-Interpreter to latest.  The most significant change is using ES6's Set (if it is available) to double the speed of serialization.

Commit:
Double speed of interpreter serialization

Commit:
Speed up serialize some more.

Commit:
Slight simplifications to 'catch' and 'with' blocks.

cpcallen commented on examples/interpreter-demo/interpreter.js:
```diff
@@ -3831,17 +3826,14 @@ Interpreter.prototype['stepCallExpression'] = function(stack, state, node) {
+  // Add the argument.
+  this.setProperty(scope.object, node['param']['name'], state.throwValue);
+  // Execute catch clause.
+  return new Interpreter.State(node['body'], scope);
```
As discussed: this can be inlined in `stepTryStatement`.

NeilFraser replied:
Done.

cpcallen commented on examples/interpreter-demo/serialize.js:
```diff
@@ -17,10 +17,21 @@ var NODE_LOC_CONSTRUCTOR;
+/**
+ * All non-primitves in the interpreter as a Set.
+ * Double the speed of serialization if ES6's Set is available.
+ * @type {Set|undefined}
```
Looks like a `Map` to me.
```suggestion
 * A Map mapping non-primitves their corresponding indexes in objectList.
 * Doubles the speed of serialisation if ES6's Map is available.
 * @type {Map|undefined}
```

NeilFraser replied:
Done.

Commit:
Update to latest JS-Interpreter

Fix Map/Set comments in serialize.
Inline CatchClause into TryStatement.
Comment out obsolete 'guard' props in Acorn.

cpcallen reviewed (approved):
L even more GTM.
